ReactApp1/ReactApp1.Server/Data/Repositories/ItemRepository.cs

```diff
@@ -0,0 +1,75 @@
+using Microsoft.EntityFrameworkCore;
+using ReactApp1.Server.Extensions;
+using ReactApp1.Server.Models;
+
+namespace ReactApp1.Server.Data.Repositories
+{
+    public class ItemRepository : IItemRepository
+    {
+        private readonly AppDbContext _context;
+
+        public ItemRepository(AppDbContext context)
+        {
+            _context = context;
+        }
+        
+        public async Task<IEnumerable<Item>> GetAllItemsAsync(int pageNumber, int pageSize)
+        {
+            return await _context.Set<Item>()
+                .OrderBy(item => item.ItemId)
+                .Paginate(pageNumber, pageSize)
+                .ToListAsync();
+        }
+
+        public async Task<Item?> GetItemByIdAsync(int itemId)
+        {
+            return await _context.Set<Item>().FindAsync(itemId);
+        }
+        
+        public async Task AddItemAsync(Item item)
+        {
+            try
+            {
+                await _context.Set<Item>().AddAsync(item);
+                await _context.SaveChangesAsync();
+
+            }   
+            catch (DbUpdateException e)
+            {
+                throw new Exception("An error occurred while adding new item to the database.", e);
+            }
+        }
+        
+        public async Task UpdateItemAsync(Item item)
+        {
+            try
+            {
+                _context.Set<Item>().Update(item);
+                await _context.SaveChangesAsync();
+
+            }   
+            catch (DbUpdateException e)
+            {
+                throw new Exception($"An error occurred while updating the item: {item.ItemId}.", e);
+            }
+        }
+        
+        public async Task DeleteItemAsync(int itemId)
+        {
+            try
+            {
+                _context.Set<Item>().Remove(new Item
```

Comment:
Lyg trinant item, reikėtų taip pat ištrinti ir "Storage" lentoje esančią eilutę, kurios ItemId yra toks pat kaip trinamo iš Item lentos

Reply:
Nors šitą gal geriau padaryti su atskiru storage servisu ir iškviesti (jeigu reikia) per item.Delete endpointą

Reply:
Dabar nesu tikras, ar Item'as Storage lentoje yra FK. Bet jei taip, tada reiktu paprasciausiai Item'ą trinant naudot DELETE CASCADE.

Reply:
Ai jo, yra DELETE CASCADE, šitą reikės pridėt tiesiog ir migracijas praleist